src/main/java/seedu/jarvis/commons/util/StringUtil.java

```diff
@@ -65,4 +65,87 @@ public static boolean isNonZeroUnsignedInteger(String s) {
             return false;
         }
     }
+
+    /**
+     * Checks if the given {@code StringBuilder} can accommodate its current length, a new
+     * token and an extra newline and space, beneath the given limit.
+     *
+     * @param sb {@code StringBuilder} to check
+     * @param token {@code String} to check
+     * @param limit length limit of the {@code String}
+     * @return true if the result {@code String} can accommodate the limit
+     */
+    private static boolean canAppend(StringBuilder sb, String token, int limit) {
+        int tokenLen = token.length();
+        int sbLen = sb.length();
+        int newLineLen = "\n".length();
+        int spaceLen = " ".length();
+        return tokenLen + sbLen + newLineLen + spaceLen <= limit;
+    }
+
+    /**
+     * Appends the given token to the given {@code StringBuilder}. This method
+     * only appends just the token if the {@code StringBuilder} is empty, the token with
+     * a preceding a space otherwise.
+     */
+    private static void appendToken(StringBuilder sb, String token) {
+        if (sb.length() == 0) {
```

Comment:
can consider abstracting `sb.length() == 0` and other similar instances in this file. Reasoning is as explained in this article https://nus-cs2103-ay1920s1.github.io/website/book/codeQuality/#avoid-magic-numbers.

Reply:
Resolved in 586cb58.